adding torrent search cog

rssnyder opened the pull request:
Add ability to search for torrents on TPB.

Usage: `!torrent last week tonight 420p`

Posts magnet link, or link to TPB if no results returned.

rssnyder committed:
adding torrent search cog

rssnyder committed:
erronious newline

Joelpls commented on cogs/torrents.py:
```diff
@@ -0,0 +1,66 @@
+
+    try:
+        response = requests.get(f'https://apibay.org/q.php?q={search}')
+    except Exception:
```
This exception should be more explicit

rssnyder replied:
Right, are we just logging via stdout?

Joelpls replied:
I don't do much logging. Don't tell anyone.

Joelpls commented on cogs/torrents.py:
```diff
@@ -0,0 +1,66 @@
+        title=target.get('name'),
+        description=desc
+    )
+    success.set_footer(text='I am not responsible for anything downloaded via this cog')
```
change cog to command.
Does it even need this as a footer tho?

rssnyder replied:
Probably not. Removed.

Joelpls commented on cogs/torrents.py:
```diff
@@ -0,0 +1,66 @@
+
+
+def get_magnet(search: str) -> discord.Embed:
+    """Get a magnet link for a torrent"""
```
Add Usage to the help comments. See https://github.com/Joelpls/discord-bot/blob/1488dca517c77d15962b16358f473c8d53e70f83/cogs/memes.py#L37

rssnyder committed:
add desc for helper method
catch specific exception, print message to the stdout logs
remove unnessesary footer

rssnyder committed:
actaully remove footer /s

Joelpls commented on cogs/torrents.py:
```diff
@@ -0,0 +1,70 @@
+
+
+def get_magnet(search: str) -> discord.Embed:
+    """
+    Get a magnet link for a torrent.
+    Uses the backend api for 'the pirate bay' to find torrent hash and name.
+    Create the magnet link based on the information returned.
+    Present to the user with helpful information.
+    """
```
I think something better is:
Get a magnet link for a torrent.
Usage: !torrent <search>

Something like that. It doesn't need all the technical details. This is what shows up when the user types !help torrent

rssnyder replied:
Ahhhhh ok makes sense.

rssnyder committed:
add user facing message

rssnyder committed:
move around documentation

rssnyder commented:
Whats the hold up, I want to start browsing!
